### PR TITLE
[ENH] sets module - skbase compatibility

### PR DESCRIPTION
**Pull request by VascoSch92**
I'm sorry. 

I did some mistake with the new and old branch. 

The easiest way was to create a new branch and open a PR. 

I will close the other ones.

### CODE REVIEW

**Comment by fkiraly** [comment]
No need to be sorry.

This is git: https://xkcd.com/1597/

File: skpro/domains/domains.py [hunk 1]
```diff
@@ -126,6 +126,13 @@ def boundary(self) -> Tuple[float, float]:
         """Return the boundary of the interval, i.e., the extremities."""
         return self._left, self._right
 
+    def get_test_params(cls, parameter_set="default"):
```
Comment (fkiraly) [inline skpro/domains/domains.py:129]: I think you are missing the `@classmethod` decorator, then it should work.

Comment (VascoSch92) [reply skpro/domains/domains.py:130]: added

Comment (fkiraly) [reply skpro/domains/domains.py:130]: restarted tests

**Comment by fkiraly** [comment]
Made a few changes to ensure `skbase` compatibility.

FYI, the API convention is that all classes must behave like dataclasses:

* init params are written to `self` in the constructor
* over the lifetime of the object, they are never mutated, only read

File: skpro/domains/domains.py [hunk 2]
```diff
@@ -126,6 +129,14 @@ def boundary(self) -> Tuple[float, float]:
         """Return the boundary of the interval, i.e., the extremities."""
         return self._left, self._right
 
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator."""
+        return {
+            "values": [1, 2],
+            "parenthesis": "open",
```
Comment (VascoSch92) [inline skpro/domains/domains.py:137]:
```suggestion
            "_parenthesis": "open",
```

**Comment by VascoSch92** [comment]
> Made a few changes to ensure `skbase` compatibility.
> 
> FYI, the API convention is that all classes must behave like dataclasses:
> 
> * init params are written to `self` in the constructor
> * over the lifetime of the object, they are never mutated, only read

Ok.

Now the testing are failing because of this error:
```
ERROR skpro/tests/test_all_estimators.py::TestAllObjects::test_repr[Product] - AttributeError: 'Interval' object has no attribute 'parenthesis'
```
You add the attribute `_parenthesis`. Therefore, i changed in the get_test_param the key `parenthesis` to `_parenthesis`.

I hope i correct :-)

**Comment by fkiraly** [comment]
Ah, that was my mistake! The `parenthesis` arg should be always what was passed to `__init__`, so the "resolve" needs to write to another arg. The `get_test_params` reference still should be `"parenthesis"`  - I fixed it and it should work now.

**Comment by VascoSch92** [comment]
Still an error but i think i know what is. I will give a look

**Comment by fkiraly** [comment]
I see, this raises an interesting question.

Do we want to enforce the dataclass like API from `BaseObject` on the init args that get simplified?

**Comment by VascoSch92** [comment]
Good question.

I don't have a strong opionion.

What are the pros to use it?

**Comment by fkiraly** [comment]
> What are the pros to use it?

Immutability of the blueprint - you can identify the object you are dealing with by the class and params uniquely. The usual case this guards against is change of the object blueprint over its lifetime - but not "simplification" operations, as in this case.

**Comment by VascoSch92** [comment]
If we want to have it, i can change the class such that it thrown an error if you are creating a finite sets with repetition. Let me know what you orefer ;)

I can change it tomorrow and start to work with distribuitions ;)

**Comment by fkiraly** [comment]
Hm, I'd be fine with either solution here. Perhaps raising an error if not unique makes most sense.

**Review by fkiraly — APPROVED** [review]
it does!

**Comment by VascoSch92** [comment]
Can we merge such that I can start with domains and distributions ? :-)

**Pull request merged**